src/ResourceManager/Profile/Commands.Profile/Errors/ResolveError.cs

```diff
@@ -0,0 +1,133 @@
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+// http://www.apache.org/licenses/LICENSE-2.0
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// ----------------------------------------------------------------------------------
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Management.Automation;
+using Microsoft.Azure.Commands.Common.Authentication.Abstractions;
+using System.Collections;
+using Microsoft.Azure.Commands.ResourceManager.Common;
+
+namespace Microsoft.Azure.Commands.Profile.Errors
+{
+    [Cmdlet(VerbsDiagnostic.Resolve, "AzureRmError", DefaultParameterSetName = ResolveError.AnyErrorParameterSet)]
+    [OutputType(typeof(AzureErrorRecord))]
+    [OutputType(typeof(AzureExceptionRecord))]
+    [OutputType(typeof(AzureRestExceptionRecord))]
+    public class ResolveError : AzureRMCmdlet
+    {
+        public const string AnyErrorParameterSet = "AnyErrorParameterSet";
+        public const string LastErrorParameterSet = "LastErrorParameterSet";
+        [Parameter(Mandatory =false, Position =0, HelpMessage ="The error records to resolve", ValueFromPipeline =true, ParameterSetName = AnyErrorParameterSet)]
+        public ErrorRecord[] Error { get; set; }
+
+        [Parameter(Mandatory = true, HelpMessage = "The last error", ParameterSetName = LastErrorParameterSet)]
+        public SwitchParameter Last { get; set; }
+        protected override IAzureContext DefaultContext
+        {
+            get
+            {
+                return null;
+            }
+        }
+
+
+        public override void ExecuteCmdlet()
+        {
+            IEnumerable<ErrorRecord> records = null;
+            base.ExecuteCmdlet();
+            switch(ParameterSetName)
+            {
+                case LastErrorParameterSet:
+                    var errors = GetErrorVariable();
+                    if (errors != null)
+                    {
+                        var error = errors.FirstOrDefault();
+                        if (error != null)
+                        {
+                            records = new ErrorRecord[] { error };
+                        }
+                    }
+                    break;
+                default:
+                    records = Error ?? GetErrorVariable();
+                    break;
+            }
+
+            if (records != null)
+            {
+                foreach (var error in records)
+                {
+                    ErrorRecord record = error as ErrorRecord;
+                    HandleError(record);
+                }
+            }
+        }
+
+        IEnumerable<ErrorRecord> GetErrorVariable()
```

Comment:
@markcowl nit: should the below methods be private if they're only being used for this cmdlet?